portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public final class BlogHashMap {
+  private static final String[] internalTags = {"#general", "#wellbeing", "#music", "#games"};
+
+  // Takes and puts a list of BlogMessage type in map.
+  // Each BlogMessage type should contain tag that matches tags in the internal System. 
+  public static void putInMap(List<BlogMessage> messages, Map<String, LinkedList<BlogMessage>> map) {
+    for (BlogMessage message : messages) {
+      boolean isTagSupported = false;
+      for (int i = 0; i < internalTags.length; i++){
+        if (message.getTag().equals(internalTags[i])){
+          isTagSupported = true;
+          if (map.containsKey(message.getTag())){
+            map.get(message.getTag()).addLast(message);
+          } else {
+            // No such mapping with specified tag,
+            // Create first mapping with specified tag.
+            LinkedList<BlogMessage> firstValue = new LinkedList<BlogMessage>();
+            firstValue.addFirst(message); 
+            map.put(message.getTag(), firstValue);
+          }
+        } 
+      }
+      if (!isTagSupported) {
+        System.out.println("Tag not supported. Tag: " + message.getTag());
+        System.out.println("For BlogMessage@ " + message);
+      }
+    }
+
+  }
+
+  public static int getSize(Map<String, LinkedList<BlogMessage>> map) {
+    return map.size();
+  }
+
+  // Takes map, and
+  // Returns a linkedList of all values of BlogMessage type in the map. 
+  public static LinkedList<BlogMessage> getMessages(Map<String, LinkedList<BlogMessage>> map) {
+    System.out.println("Fetching all [LinkedList] Values in map...");
+
+    Set<String> keys = map.keySet();
+    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
+    for (String key : keys) {
+      values.addAll(map.get(key));
```

Comment:
What is supposed to be included in the returned linked list? There seems to be duplicate messages, would that be a problem? How would this be used?

If those are all well thought out, please put that in the method comment as well because it doesn't seem straightforward and worth extra expalantion.

Reply:
Since each Blogmessage has one tag I don't think there'll be duplicate messages for a single tag. There might be same messages for different tags but that will not be a problem. I don't know if that clarifies what you addressed in the comment

Reply:
That clarifies it. That was what I thought (one message can have multiple tags) too.

In this case, I think it would be good to add a comment in the method saying "The returned list may contain duplicated messages" (just as an FYI for callers so that they don't accidentally think this is a list of unique items).